packages/admin/src/providers/dataProvider.js

```diff
@@ -98,4 +91,25 @@ export default {
 			method: 'DELETE',
 		}).then(({ json }) => ({ data: json }));
 	},
+	upload: (resource, params) => {
+		const { data } = params;
+		const { meta = false, files } = data;
+
+		const formData = new FormData();
+
+		files.map((file, index) => {
+			return formData.append(`files[${index}]`, file);
+		});
+
+		if (meta) {
+			formData.append('meta', JSON.stringify(meta));
+		}
+
+		return httpClient(`${apiUrl}/${resource}`, {
+			method: 'POST',
+			body: formData,
+		}).then(({ json }) => ({
+			data: { ...json, id: json[0].object_id },
+		}));
```

Comment:
```suggestion
		const json = await httpClient(`${apiUrl}/${resource}`, {
			method: 'POST',
			body: formData,
		});

		return {
			data: {
				...json,
				id: json[0].object_id
			}
		}
```

Reply:
Só copiei da doc. https://marmelab.com/react-admin/doc/3.1/Tutorial.html#connecting-to-a-real-api
Posso ver isso refatorando todos os métodos em uma próxima PR